### ISSUE TITLE
Standalone validation check failed since different managers use different memory dictionary

### ISSUE THREAD

**Issue by RemindD** [eclipse-symphony/symphony#445]
For example, solutionContainer record only exists in the memory provider of solutionContainer manager. If the solution manager wants to check if the container exists prior to the creation of a solution, it needs to call symphony API endpoints to query the solution container. 
The validation is failed currently since the solution manager cannot access the memory dictionary of solution container.
There are two potential fixes:
1. change all the query function to call symphony API endpoint
     main cost: change all the list function in the symphony client to support label filter. And different function names will make it difficult for abstraction
2. change the memory used by all memory state provider to be a global memory.
    main cost: need to manager memory state for objects with different types in a common place so need to deal with locks carefully.

In short term, [P0] consider marking the NeedValidation flag in managers to be always false.

**Comment by msftcoderdjw** [comment]
We consider to build shared memory map.
